srtcore/sync.h

```diff
@@ -267,6 +267,57 @@ class InvertedLock
     }
 };
 
+////////////////////////////////////////////////////////////////////////////////
+//
+// CCondVar section
+//
+////////////////////////////////////////////////////////////////////////////////
+
+class CCondVar
+{
+public:
+    CCondVar();
+    ~CCondVar();
+
+public:
+    /// Atomically releases lock, blocks the current executing thread, 
+    /// and adds it to the list of threads waiting on *this.
+    /// The thread will be unblocked when notify_all() or notify_one() is executed,
+    /// or when the relative timeout rel_time expires.
+    /// It may also be unblocked spuriously. When unblocked, regardless of the reason,
+    /// lock is reacquired and wait_for() exits.
+    ///
+    /// @returns false if the relative timeout specified by rel_time expired,
+    ///          true otherwise (signal or spurious wake up).
+    ///
+    /// @note Calling this function if lock.mutex()
+    /// is not locked by the current thread is undefined behavior.
+    /// Calling this function if lock.mutex() is not the same mutex as the one
+    /// used by all other threads that are currently waiting on the same
+    /// condition variable is undefined behavior.
+    bool wait_for(UniqueLock& lock, const steady_clock::duration& rel_time);
```

Comment:
Question: is wait_until agreed to be implemented only in the `CSync` class? Still acceptable, but maybe, you know.

Reply:
Added.